src/plugins/discover/public/application/apps/main/components/doc_table/components/pager/tool_bar_pagination.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPagination,
+  EuiPopover,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n/';
+
+interface ToolBarPaginationProps {
+  pageSize: number;
+  pageCount: number;
+  activePage: number;
+  onPageClick: (page: number) => void;
+  onPageSizeChange: (size: number) => void;
+}
+
+export const ToolBarPagination = ({
+  pageSize,
+  pageCount,
+  activePage,
+  onPageSizeChange,
+  onPageClick,
+}: ToolBarPaginationProps) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const rowsWord = i18n.translate('discover.docTable.rows', {
+    defaultMessage: 'rows',
+  });
+
+  const onChooseRowsClick = () => setIsPopoverOpen((prevIsPopoverOpen) => !prevIsPopoverOpen);
+
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const getIconType = (size: number) => {
+    return size === pageSize ? 'check' : 'empty';
+  };
+
+  const rowsPerPageOptions = [25, 50, 100].map((cur) => (
+    <EuiContextMenuItem
+      key={`${cur} rows`}
+      icon={getIconType(cur)}
+      onClick={() => {
+        closePopover();
+        onPageSizeChange(cur);
+      }}
+    >
+      {cur} {rowsWord}
```

Comment:
this is bad practice from localization point of view, as the word itself could be different depending on the number (singular / plural / multiple variations of plurals in some languages). for this use-case it's fine, just something to keep in mind.

Reply:
Thank you!